nixos/modules/system/etc/setup-etc.pl

```diff
@@ -85,62 +94,78 @@ sub cleanup {
+    my $fn = substr($File::Find::name, length($etc) + 1) or return;
     my $target = "/etc/$fn";
-    File::Path::make_path(dirname $target);
+    File::Path::make_path(dirname($target));
     $created{$fn} = 1;
 
     # Rename doesn't work if target is directory.
-    if (-l $_ && -d $target) {
-        if (isStatic $target) {
-            rmtree $target or warn;
+    if (-l $path_to_file && -d $target) {
+        if (is_static($target)) {
+            rmtree($target) or warn("Failed to remove $target");
         } else {
-            warn "$target directory contains user files. Symlinking may fail.";
+            warn("$target directory contains user files. Symlinking may fail.");
         }
     }
 
-    if (-e "$_.mode") {
-        my $mode = read_file("$_.mode"); chomp $mode;
+    # This will set uid/gid if the options were set in the nix config
+    if (-e "$path_to_file.mode") {
+        chomp(my $mode = read_file("$path_to_file.mode"));
         if ($mode eq "direct-symlink") {
-            atomicSymlink readlink("$static/$fn"), $target or warn;
+            atomic_symlink(readlink("$static/$fn"), $target) or warn;
         } else {
-            my $uid = read_file("$_.uid"); chomp $uid;
-            my $gid = read_file("$_.gid"); chomp $gid;
-            copy "$static/$fn", "$target.tmp" or warn;
-            $uid = getpwnam $uid unless $uid =~ /^\+/;
-            $gid = getgrnam $gid unless $gid =~ /^\+/;
-            chown int($uid), int($gid), "$target.tmp" or warn;
-            chmod oct($mode), "$target.tmp" or warn;
-            rename "$target.tmp", $target or warn;
+            chomp(my $uid = read_file("$path_to_file.uid"));
+            chomp(my $gid = read_file("$path_to_file.gid"));
+            copy("$static/$fn", "$target.tmp") or warn;
+
+            # uid could either be an uid or an username, this gets the uid
+            if ($uid !~ /^\+/msx) {
+                $uid = getpwnam($uid);
+            }
+            if ($gid !~ /^\+/msx) {
+                $gid = getgrnam($gid);
+            }
+
+            chown(int($uid), int($gid), "$target.tmp") or warn;
+            chmod(oct($mode), "$target.tmp") or warn;
+            rename("$target.tmp", $target) or warn;
         }
-        push @copied, $fn;
-        print CLEAN "$fn\n";
-    } elsif (-l "$_") {
-        atomicSymlink "$static/$fn", $target or warn;
+        push(@copied, $fn);
+    } elsif (-l "$path_to_file") {
+        atomic_symlink("$static/$fn", $target) or warn;
     }
+    return;
 }
 
-find(\&link, $etc);
+find(\&make_symlinks, $etc);
 
 
 # Delete files that were copied in a previous version but not in the
 # current.
-foreach my $fn (@oldCopied) {
+foreach my $fn (@old_copied) {
     if (!defined $created{$fn}) {
         $fn = "/etc/$fn";
-        print STDERR "removing obsolete file ‘$fn’...\n";
-        unlink "$fn";
+        print(STDERR "removing obsolete file ‘$fn’...\n");
+        unlink("$fn") or warn("Couldn't remove $fn");
```

Comment:
```suggestion
        unlink("$fn") or warn("Couldn't remove $fn: $!");
```

- Add `$!` to get the error
- Consider `die`